zetaclient/chains/evm/signer/gas.go

```diff
@@ -0,0 +1,138 @@
+package signer
+
+import (
+	"fmt"
+	"math/big"
+
+	"cosmossdk.io/math"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog"
+
+	"github.com/zeta-chain/zetacore/x/crosschain/types"
+)
+
+const (
+	minGasLimit = 100_000
+	maxGasLimit = 1_000_000
+)
+
+// Gas represents gas parameters for EVM transactions.
+//
+// This is pretty interesting because all EVM chains now support EIP-1559, but some chains do it in a specific way
+// https://eips.ethereum.org/EIPS/eip-1559
+// https://www.blocknative.com/blog/eip-1559-fees
+// https://github.com/bnb-chain/BEPs/blob/master/BEPs/BEP226.md (tl;dr: baseFee is always zero)
+//
+// However, this doesn't affect tx creation nor broadcasting
+type Gas struct {
+	limit uint64
+
+	// This is a "total" gasPrice per 1 unit of gas.
+	price *big.Int
+
+	// PriorityFee a fee paid directly to validators.
+	priorityFee *big.Int
+}
+
+func (g Gas) validate() error {
```

Comment:
Shouldn't we consider `g.price > g.priorityFee` as part of the check?

Reply:
https://github.com/zeta-chain/node/pull/2634/commits/086c9af4236a1a0cf6e846e7e6df933c9fe88b15